### PR TITLE
chore: update browserlist and babel targets to chrome/firefox 89

### PR DESCRIPTION
**Pull request by davidmurdoch**
## **Description**

IIRC, we've documented extension browser support for versions 2 years old and newer (I can't find this documentation now though). Chrome and Firefox 89 were released in 2021, so the version in this PR should be good to go.

We could go with even newer browser versions, but it was a struggle getting it to work for v89, so I didn't try to take the versions any further.

This PR also reduces the chance of seeing a lavamoat caused `RegExpCompiler Allocation failed - process out of memory` error (https://github.com/MetaMask/metamask-extension/issues/21006).

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .browserslistrc [hunk 1]
```diff
@@ -1 +1 @@
-chrome >= 80, firefox >= 78
+chrome >= 89, firefox >= 89
```
Comment (Gudahtt) [inline .browserslistrc:1]: Looks like these versions are below the `OUTDATED_BROWSER_VERSIONS` we have configured, so this looks good :+1:

File: .yarn/patches/lavamoat-core-npm-14.4.1-c4e8bbb016.patch [hunk 2]
```diff
@@ -1,31 +0,0 @@
-diff --git a/src/kernelTemplate.js b/src/kernelTemplate.js
-index de8d73048b7bb7c4e74009b5c85ad919fe197ef0..e5bcb987cff013def8b2e8c767eb75a57ea0bd7a 100644
---- a/src/kernelTemplate.js
-+++ b/src/kernelTemplate.js
-@@ -60,6 +60,8 @@
-       errorTaming: 'unsafe',
-       // shows the full call stack
-       stackFiltering: 'verbose',
-+      // prevents most common override mistake cases from tripping up users
-+      overrideTaming: 'severe',
```
Comment (Gudahtt) [inline .yarn/patches/lavamoat-core-npm-14.4.1-c4e8bbb016.patch:10]: Is this patch still relevant to v15?

Comment (Gudahtt) [reply .yarn/patches/lavamoat-core-npm-14.4.1-c4e8bbb016.patch:10]: Ah, I see that this is line is in v15 already

File: package.json [hunk 3]
```diff
@@ -189,7 +189,6 @@
     "@eslint/eslintrc@^2.0.1": "patch:@eslint/eslintrc@npm%3A2.0.2#./.yarn/patches/@eslint-eslintrc-npm-2.0.2-d308674d86.patch",
     "@fortawesome/fontawesome-free@^5.13.0": "patch:@fortawesome/fontawesome-free@npm%3A5.13.0#./.yarn/patches/@fortawesome-fontawesome-free-npm-5.13.0-f20fc0388d.patch",
     "@keystonehq/bc-ur-registry@^0.5.0-alpha.5": "patch:@keystonehq/bc-ur-registry@npm%3A0.5.0-alpha.5#./.yarn/patches/@keystonehq-bc-ur-registry-npm-0.5.0-alpha.5-b95c7992a6.patch",
-    "@lavamoat/lavapack@^3.1.0": "patch:@lavamoat/lavapack@npm%3A3.1.0#./.yarn/patches/@lavamoat-lavapack-npm-3.1.0-34c65d233b.patch",
```
Comment (Gudahtt) [inline package.json:192]: Interesting, this patch was needed to remove absolute file paths from the bundle, but it looks like that was fixed upstream (relative file paths are used instead). Nice, one less patch needed!

Could you remove the patch file as well? It looks like it's still present in the `.yarn/patches` directory

Comment (Gudahtt) [reply package.json:192]: I have just double checked to confirm that the absolute paths are gone. They're all relative paths now, so this patch is definitely safe to remove.

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Great work! We need to preserve one part of the `lavamoat-core` patch that was removed, and there is an extra patch file to delete for lavapack. Everything looks good aside from that.

File: babel.config.js [hunk 4]
```diff
@@ -5,8 +5,15 @@ module.exports = function (api) {
...
     targets: {
-      browsers: ['chrome >= 80', 'firefox >= 78'],
+      browsers: ['chrome >= 89', 'firefox >= 89'],
     },
+    plugins: [
+      // `browserify` is old and busted, and doesn't support `??=` (and other
+      // logical assignment operators). This plugin lets us target es2020-level
+      // browsers (except we do still end up with transpiled logical assignment
+      // operators 😭)
+      '@babel/plugin-transform-logical-assignment-operators',
```
Comment (Gudahtt) [inline babel.config.js:15]: We'll need to list this in the depcheck config, as depcheck doesn't see where it's used

File: .yarn/patches/lavamoat-core-npm-14.4.1-c4e8bbb016.patch [hunk 5]
```diff
@@ -1,31 +0,0 @@
...
-     }
-     const policyOverride = await readPolicyFile({ debugMode, policyPath: policyOverridePath })
-     lavamoatPolicy = mergePolicy(policy, policyOverride)
--    // TODO: Only write if merge results in changes.
--    // Would have to make a deep equal check on whole policy, which is a waste of time.
--    // mergePolicy() should be able to do it in one pass.
--    fs.writeFileSync(policyPath, jsonStringify(lavamoatPolicy, { space: 2 }))
-+    // Skip policy write step to prevent intermittent build failures
-+    // The extension validates the policy in a separate step, we don't need it
-+    // to be written to disk here.
```
[the remark this answers is not in the document]
Comment (davidmurdoch) [reply .yarn/patches/lavamoat-core-npm-14.4.1-c4e8bbb016.patch:28]: Fixed! Good catch. I think I just ran into this bug a few minutes ago on another PR that depends on this one.

(Patching an existing yarn patch is absolute torture.)

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by davidmurdoch** [comment]
@Gudahtt I didn't see your review and comment till after I commit a fixed to the policies. Sorry!

**Review by Gudahtt — DISMISSED** [review]

**Review by FrederikBolding — DISMISSED** [review]

**Comment by dbrans** [comment]
> This PR also reduces the chance of seeing a lavamoat caused RegExpCompiler Allocation failed - process out of memory error (https://github.com/MetaMask/metamask-extension/issues/21006).

Out of curiosity, what is the connection between the browser version and the out of memory error?

**Comment by davidmurdoch** [comment]
@dbrans @Gudahtt This already occurs at the current browser version levels, just not as frequently. Bumping the browser version seems to have caused additional regex memory usage.

I'm created a PR with the LavaMoat patches upstream: https://github.com/LavaMoat/LavaMoat/pull/867

**Comment by Gudahtt** [comment]
@davidmurdoch is this ready to merge? I'm seeing the OOM issue on `develop` fairly often, so I'm eager to see if these patch solves the issue or not.

**Comment by davidmurdoch** [comment]
@Gudahtt nothing blocking on my end. Hopefully the additions to the lavamoat patches will be short-lived!

**Review by legobeat — DISMISSED** [review]
LGTM, though it needs a sync with `develop` and resulting conflict resolution before merging

**Comment by Gudahtt** [comment]
Looks like the policy update failed due to a syntax error in the policy file, which the update script isn't tolerant of. I'd recommend resetting the policies to match `develop`, then retrying the update-policies command

**Comment by Gudahtt** [comment]
Thanks, sounds good, I had thought the policy check was failing when I made that comment but I must have been looking at the wrong workflow.

File: package.json [hunk 6]
```diff
@@ -218,7 +218,7 @@
     "@metamask/eth-keyring-controller@npm:^13.0.1": "patch:@metamask/eth-keyring-controller@npm%3A13.0.1#~/.yarn/patches/@metamask-eth-keyring-controller-npm-13.0.1-06ff83faad.patch",
     "nonce-tracker@npm:^3.0.0": "patch:nonce-tracker@npm%3A3.0.0#~/.yarn/patches/nonce-tracker-npm-3.0.0-c5e9a93f9d.patch",
     "@trezor/connect-web": "9.0.11",
-    "lavamoat-core@npm:^15.1.1": "patch:lavamoat-core@npm%3A15.1.1#~/.yarn/patches/lavamoat-core-npm-15.1.1-51fbe39988.patch"
+    "lavamoat-core@npm:^15.0.0": "patch:lavamoat-core@npm%3A15.0.0#~/.yarn/patches/lavamoat-core-npm-15.0.0-5f5fd568f9.patch"
```
Comment (Gudahtt) [inline package.json:221]: It looks like this patch isn't getting applied (v15.1.1 is used in the lockfile). I think this explains the intermittent policy parse issues we're seeing. That intermittent policy parse issue is what the "Skip policy write step to prevent intermittent build failures" step of the patch is meant to fix

Comment (Gudahtt) [reply package.json:221]: As an aside, this is not the first time I've noticed an unused patch go unnoticed. We should find a way to flag that in a CI check. We shouldn't have unused patches.

Comment (Gudahtt) [reply package.json]: Oh, now the whole patch is gone. The regex optimizations need to be preserved

Comment (davidmurdoch) [reply package.json]: Patching patch files is zero fun. Haha

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by brad-decker — APPROVED** [review]
